docs/operate/control/headless-app.md

```diff
@@ -5,5 +5,532 @@ weight: 30
+1. [Debug](#debug)
+
+{{% /alert %}}
+
+## Decide where to run your code
+
+You can run your code directly on the machine's single-board computer (SBC), or you can run it from a separate computer connected to the internet or to the same local network as your machine's SBC or microcontroller.
+
+### On a separate computer
+
+We recommend running your script on a laptop, desktop, or server if:
+
+- You are using computationally-intensive programs involving, for example, computer vision or motion planning, and
+- You have a stable internet connection
+
+The client code will establish a connection to the instance of `viam-server` on your machine's SBC over [LAN or WAN](/dev/reference/sdks/connectivity/).
+
+### On the machine itself
+
+We recommend running your script on the SBC that directly controls your hardware if:
+
+- Your machines have intermittent or no network connectivity, or
+- You want to reduce latency, for example for running [PID control loops](https://en.wikipedia.org/wiki/Proportional%E2%80%93integral%E2%80%93derivative_controller), or
+- Your machine runs continuously (for example, an air quality sensor) and it is impractical to constantly run a client from your development computer
+
+## Install an SDK
+
+Install your preferred Viam SDK on the Linux or macOS computer SBC where you plan to run the control script.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+If you are using the Python SDK, [set up a virtual environment](/sdks/python/python-venv/) to package the SDK inside before running your code, avoiding conflicts with other projects or your system.
+
+For macOS (both Intel `x86_64` and Apple Silicon) or Linux (`x86`, `aarch64`, `armv6l`), run the following commands:
+
+```sh {class="command-line" data-prompt="$"}
+python3 -m venv .venv
+source .venv/bin/activate
+pip install viam-sdk
+```
+
+Windows is not supported.
+If you are using Windows, use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and install the Python SDK using the preceding instructions for Linux.
+For other unsupported systems, see [Installing from source](https://python.viam.dev/#installing-from-source).
+
+If you intend to use the [ML (machine learning) model service](/services/ml/), use the following command instead, which installs additional required dependencies along with the Python SDK:
+
+```sh {class="command-line" data-prompt="$"}
+pip install 'viam-sdk[mlmodel]'
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+Run the following command to install the [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk):
+
+```sh {class="command-line" data-prompt="$"}
+go get go.viam.com/rdk/robot/client
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+Follow the [instructions on the GitHub repository](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Authenticate your script
+
+To connect your code to your machine, use the API keys Viam automatically generates for you.
```

Comment:
I think we want to think more general. It could be the case that this headless app manages multiple machines that are maybe working in collaboration. In that case you may want to authenticate not with the API keys for a machine but with the API keys for a location or organization. See how you like this:

```suggestion
You can find sample connection code on a machine's **CONNECT** tab in the [Viam app](link):
```